cmd/rookflex/cmd/mount.go

```diff
@@ -290,5 +295,66 @@ func mountCephFS(client *rpc.Client, opts *flexvolume.AttachOptions) error {
 		log(client, fmt.Sprintf("ceph filesystem %s has been attached and mounted", opts.FsName), false)
 	}
 
-	return err
+	setFSGroup(client, opts)
+	return nil
+}
+
+// setFSGroup will set the volume ownership to the fsGroup requested in the security context of the pod mounting the storage.
+// If no fsGroup is specified, does nothing.
+// If the operation fails, the error will be logged, but will not undo the mount.
+// Follows the pattern set by the k8s volume ownership as found in
+// https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/pkg/volume/volume_linux.go#L38.
+func setFSGroup(client *rpc.Client, opts *flexvolume.AttachOptions) {
+	if opts.FsGroup == "" {
+		log(client, "no fsgroup to set for the filesystem", false)
+		return
+	}
+
+	fsGroup, err := strconv.Atoi(opts.FsGroup)
+	if err != nil {
+		log(client, fmt.Sprintf("invalid fsgroup %s. %+v", opts.FsGroup, err), true)
+		return
+	}
+
+	path := path.Join(opts.MountDir, opts.Path)
+	log(client, fmt.Sprintf("setting fsgroup %d on path %s", fsGroup, path), false)
+
+	info, err := os.Stat(path)
```

Comment:
instead of just getting the current user to set it with the fsGroup below, we should also handle the `runAsUser` that might be set in the security context next to the fsGroup.

Reply:
@travisn rather than getting the current `uid`, couldn't you just pass in `-1` to [indicate no change](https://golang.org/pkg/os/#Chown)? Setting only the `gid` seems like a better fit for CephFS volumes that may be mounted by multiple pods using the same `fsGroup` but different `runAsUser` values.

Reply:
@travisn Kubernetes does not set the uid on the mount, only the group AFAIK.

Reply:
cc @fabianvf for openshift

Reply:
Group ownership should be sufficient for OpenShift as well